hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapGetAllMessageTask.java

```diff
@@ -53,6 +55,22 @@ public String getServiceName() {
         return MapGetAllCodec.decodeRequest(clientMessage);
     }
 
+    @Override
+    protected void beforeProcess() {
+        startTimeNanos = System.nanoTime();
+    }
+
+    @Override
+    protected void beforeResponse() {
+        final long latencyNanos = System.nanoTime() - startTimeNanos;
+        final MapService mapService = getService(MapService.SERVICE_NAME);
+        MapContainer mapContainer = mapService.getMapServiceContext().getMapContainer(parameters.name);
```

Comment:
same thread visibility issue may occur here with `parameters.name` and `parameters.keys.size()` below?

Reply:
These values are created before the thread is started and does not change after that. They are immutable.
So they should be ok.